sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -256,20 +256,21 @@ private[hive] trait HiveStrategies {
    */
   object HiveTableScans extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
-      case ScanOperation(projectList, predicates, relation: HiveTableRelation) =>
+      case ScanOperation(projectList, filters, relation: HiveTableRelation) =>
         // Filter out all predicates that only deal with partition keys, these are given to the
         // hive table scan operator to be used for partition pruning.
         val partitionKeyIds = AttributeSet(relation.partitionCols)
-        val (pruningPredicates, otherPredicates) = predicates.partition { predicate =>
-          !predicate.references.isEmpty &&
-          predicate.references.subsetOf(partitionKeyIds)
-        }
+        val normalizedFilters = DataSourceStrategy.normalizeExprs(
+          filters.filter(_.deterministic), relation.output)
+
+        val partitionKeyFilters = DataSourceStrategy.getPushedDownFilters(relation.partitionCols,
+          normalizedFilters)
 
         pruneFilterProject(
           projectList,
-          otherPredicates,
+          filters.filter(f => f.references.isEmpty || !f.references.subsetOf(partitionKeyIds)),
```

Comment:
`f.references.isEmpty` means it's true or false literal?

Reply:
or a scalar subquery?

Reply:
In `FileSourceStrategy`, scalar subquery is put in partition filters, not data filters, shall we follow it?

Reply:
I am actually just following the original behavior of HiveStrategies,

regarding scalar subquery, is it in partition filter , I missed something in `FileSourceStrategy `?

Reply:
It's OK for now to be the same as before, but it's better to make it consistent with `FileSourceStrategy`. We can do it in followup.